napalm_yang/parser.py

```diff
@@ -25,29 +25,27 @@ def __init__(self, model, device=None, profile=None, is_config=None,
         self.keys = keys or {"parent_key": None}
         self.extra_vars = extra_vars or {}
 
+        if self.mapping:
+            self.parser = helpers.get_parser(self.mapping.get("metadata", {}).get("parser", ''))
```

Comment:
This is an implementation detail but I prefer the old logic: `self.mapping["metadata"]["parser"]` as that is mandatory. One module might be parsed in text mode while another in json mode so we need to define this in each module.